workspaces/core/src/combobox/Combobox.tsx

```diff
@@ -1,180 +1,546 @@
+        className)}
+      ref={popper?.setAnchor}
+    >
+      {children}
+    </div>
+  );
+};
+
+
+const Input = ({
+  displayValue,
+  placeholder,
+  type,
+  className,
+  onChange,
+  onQueryChange,
+  label,
+  ...rest
+}: InputProps) => {
+
+  const { size, popper, disabled, isError, input } = useComboboxContext('Combobox.Input');
+
+  return (
+    <HeadlessCombobox.Input
+      onChange={({ target: { value } }) => onQueryChange(value)}
+      ref={popper?.setAnchor}
+      as={NativeInput}
+      displayValue={displayValue}
+      placeholder={placeholder}
+      type={type ? type : 'text'}
+      className={mergeClassnames(
+        'flex-grow h-full border-0 !rounded-none bg-transparent px-0',
+        '!shadow-none hover:shadow-none focus:shadow-none focus-visible:shadow-none',
+        getTextSizes(size),
+        className
+      )}
+      disabled={disabled}
+      isError={isError}
+      onFocus={() => input?.setIsFocused(true)}
+      onBlur={() => input?.setIsFocused(false)}
+      {...rest}
+    />
+  );
+};
+
+const InsetInput = ({
+  displayValue,
+  placeholder,
+  type,
+  className,
+  onChange,
+  onQueryChange,
+  label,
+  ...rest
+}: InputProps) => {
+
+  const { size, popper, disabled, isError, input } = useComboboxContext('Combobox.InsetInput');
+
+  return (
+    <span className={mergeClassnames(
+      'relative',
+      'flex flex-grow w-full'
+    )}
+    >
+      <HeadlessCombobox.Input
+        onChange={({ target: { value } }) => onQueryChange(value)}
+        ref={popper?.setAnchor}
+        as={NativeInput}
+        displayValue={displayValue}
+        placeholder={placeholder === undefined ? '' : `${placeholder}`}
+        type={type ? type : 'text'}
         disabled={disabled}
+        className={mergeClassnames(
+          'flex-grow h-full border-0 !rounded-none bg-transparent px-0',
+          '!shadow-none hover:shadow-none focus:shadow-none focus-visible:shadow-none',
+          (label !== undefined && label.length > 0)
+          && (placeholder === undefined || placeholder.length === 0)
+          && 'input-xl',
+          (label !== undefined && label.length > 0)
+          && 'pt-3 input-xl-dt-label',
+          getTextSizes(size),
+          className
+        )}
+        isError={isError}
+        onFocus={() => input?.setIsFocused(true)}
+        onBlur={() => input?.setIsFocused(false)}
+        {...rest}
+      />
+      <InputInset.Label className='w-auto -top-0.5 !right-0 !left-0 whitespace-nowrap overflow-x-hidden'>{label}</InputInset.Label>
+    </span>
+  );
+};
+
+const Button = ({
+  open,
+  children,
+  label,
+  className,
+  ...rest
+}: WithChildren<ButtonProps>) => {
+  const { size, disabled } = useComboboxContext('Combobox.Button');
+
+  return (
+    <HeadlessCombobox.Button
+      className={mergeClassnames(
+        'w-6 h-6',
+        size === 'sm' ? 'text-moon-16' : 'text-moon-24',
+        size === 'sm' && 'w-4 h-4',
```

Comment:
perhaps it makes sense to combine in one condition, like: 
`size === 'sm' ? 'w-4 h-4 text-moon-16' : 'text-moon-24',`